docs/running-on-kubernetes.md

```diff
@@ -80,13 +99,45 @@ In the above example, the specific Kubernetes cluster can be used with spark sub
+disk is discussed below.
+
+## Dependency Management
+
+Application dependencies that are being submitted from your machine need to be sent to a **resource staging server**
+that the driver and executor can then communicate with to retrieve those dependencies. A YAML file denoting a minimal
+set of Kubernetes resources that runs this service is located in the file `conf/kubernetes-resource-staging-server.yaml`.
+This YAML file configures a Deployment with one pod running the resource staging server configured with a ConfigMap,
+and exposes the server through a Service with a given NodePort.
+
+To run the resource staging server with default configurations, the Kubernetes resources can simply be created:
+
+    kubectl create -f conf/kubernetes-resource-staging-server.yaml
+
+and then you can compute the value of Pi as follows:
+
+    bin/spark-submit \
+      --deploy-mode cluster \
+      --class org.apache.spark.examples.SparkPi \
+      --master k8s://https://<k8s-apiserver-host>:<k8s-apiserver-port> \
+      --kubernetes-namespace default \
+      --conf spark.executor.instances=5 \
+      --conf spark.app.name=spark-pi \
+      --conf spark.kubernetes.driver.docker.image=kubespark/spark-driver:v2.1.0-kubernetes-0.1.0-alpha.2 \
+      --conf spark.kubernetes.executor.docker.image=kubespark/spark-executor:v2.1.0-kubernetes-0.1.0-alpha.2 \
+      --conf spark.kubernetes.initcontainer.docker.image=kubespark/spark-init:v2.1.0-kubernetes-0.1.0-alpha.2 \
+      --conf spark.kubernetes.resourceStagingServer.uri=http://<address-of-any-cluster-node>:31000 \
+      examples/jars/spark_examples_2.11-2.2.0.jar
+
+The Docker image for the resource staging server may also be built from source, in a similar manner to the driver
+and executor images. The Dockerfile is provided in `dockerfiles/resource-staging-server/Dockerfile`.
 
-Spark supports specifying JAR paths that are either on the submitting host's disk, or are located on the disk of the
-driver and executors. Refer to the [application submission](submitting-applications.html#advanced-dependency-management)
-section for details. Note that files specified with the `local://` scheme should be added to the container image of both
-the driver and the executors. Files without a scheme or with the scheme `file://` are treated as being on the disk of
-the submitting machine, and are uploaded to the driver running in Kubernetes before launching the application.
+Note that this resource staging server is only required for submitting local dependencies. If your application's
+dependencies are all hosted in remote locations like HDFS or http servers, they may be referred to by their appropriate
+remote URIs. Also, application dependencies can be pre-mounted into custom-built Docker images. Those dependencies
+can be added to the classpath by referencing them with `local://` URIs and/or setting the `SPARK_EXTRA_CLASSPATH`
```

Comment:
It basically is anything that can be passed to `-cp` in a Java invocation - see [this](https://github.com/apache-spark-on-k8s/spark/blob/branch-2.1-kubernetes/resource-managers/kubernetes/docker-minimal-bundle/src/main/docker/driver-v2/Dockerfile#L42) on how we use it.